config/config.go

```diff
@@ -163,31 +169,12 @@ func (c *Config) Save() error {
 
 // Set preserve sd-local config with new value.
 func (e *Entry) Set(key, value string) error {
-	switch key {
-	case "api-url":
-		e.APIURL = value
-	case "store-url":
-		e.StoreURL = value
-	case "token":
-		e.Token = value
-	case "launcher-version":
-		if value == "" {
-			value = "stable"
-		}
-		e.Launcher.Version = value
-	case "launcher-image":
-		if value == "" {
-			value = "screwdrivercd/launcher"
-		}
-		e.Launcher.Image = value
-	case "uuid":
-		if value == "" {
-			value = "-"
-		}
-		e.UUID = value
-	default:
+	var m map[string]interface{}
+	mapstructure.Decode(e, &m)
+	if _, ok := m[key]; !ok {
 		return fmt.Errorf("invalid key %s", key)
 	}
-
+	m[key] = value
+	mapstructure.Decode(m, &e)
```

Comment:
This code should work on update value (except empty value handling), although I suggest add some comments to explain how works.
1. encode current configuration to empty map
2. check map key found (error handling for unknown key) and set value
3. update current configuration by map